front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/PipelineTemplateController.java

```diff
@@ -105,29 +119,57 @@ void checkForDependentConfigs(String templateId) {
           return;
         }
 
-        if (source != null && source.equalsIgnoreCase("spinnaker://" + templateId)) {
+        if (source != null && containsAnyIgnoreCase(source, templateIds)) {
           dependentConfigIds.add(templatedPipeline.getId());
         }
       });
+    return dependentConfigIds;
+  }
+
+  private List<String> convertAllTemplateIdsToSources(String rootTemplateId) {
+    List<String> templateIds = new ArrayList<>();
+    templateIds.add("spinnaker://" + rootTemplateId);
+    for (String id : getDependentTemplates(rootTemplateId)) {
+      templateIds.add("spinnaker://" + id);
+    }
+    return templateIds;
+  }
+
+  private static boolean containsAnyIgnoreCase(String source, List<String> templateIds) {
+    for (String templateId : templateIds) {
+      if (source.equalsIgnoreCase(templateId)) {
+        return true;
+      }
+    }
+    return false;
+  }
 
+  @VisibleForTesting
+  void checkForDependentConfigs(String templateId) {
+    List<String> dependentConfigIds = getDependentConfigs(templateId);
     if (dependentConfigIds.size() != 0) {
       throw new InvalidRequestException("The following pipeline configs"
         + " depend on this template: " + String.join(", ", dependentConfigIds));
     }
   }
 
   @VisibleForTesting
-  void checkForDependentTemplates(String templateId) {
+  List<String> getDependentTemplates(String templateId) {
     List<String> dependentTemplateIds = new ArrayList<>();
-
     getPipelineTemplateDAO().all()
```

Comment:
This might get real slow given the recursive call...

Reply:
That was one of my concerns as well. I think I'll make it optional via query param

Reply:
I think you should be able to loop just twice over the set of templates. The first time you would build a map `templateId -> sourceId`. The second time, for each template, you would walk down the map (`A -> B -> C -> ...`) for each template until you reach a root template.